### PR TITLE
Improve buffer handling in regex match processing

### PR DESCRIPTION
**Pull request by vikman90**
## Description

This PR enhances the robustness of string handling in the `match_regex()` function used for parsing compliance tags in rootcheck events. The changes add proper bounds checking and safe string operations to ensure reliable processing of variable-length input strings.

### Code Changes

**File: `src/analysisd/format/json_extended.c`**

1. **Array bounds validation**: Added check before processing matches to prevent exceeding the `MAX_MATCHES` limit (10 results)
2. **Length validation**: Introduced match length validation before buffer write operations
3. **Safe string copy**: Replaced `sprintf()` with `snprintf()` for bounded writes
4. **Defensive null termination**: Added explicit null termination to ensure valid strings
5. **Post-increment validation**: Added additional bounds check after incrementing counter for defense in depth

#### Before changes
- Function used unbounded `sprintf()` which could write beyond allocated buffer size
- No validation of match length before write operations
- No explicit handling of oversized compliance tags

#### After changes
- All string writes are bounded by buffer size (30 bytes)
- Oversized matches are safely truncated with logging
- Explicit null termination guarantees valid strings
- Array bounds are validated before access

#### Logs

Example log output when processing oversized tags:**
```
2026/03/24 17:18:09 wazuh-analysisd[8379] json_extended.c:641 at match_regex(): DEBUG: Match length (102) exceeds buffer size, truncating to 29.
```

### Artifacts Affected

- **wazuh-analysisd**: Core analysis daemon binary

### Configuration Changes

No configuration changes required. The improvements are transparent to existing deployments.

### Documentation Updates

No user-facing documentation updates required.

### Tests Introduced

**New test file:** `src/unit_tests/analysisd/test_json_extended.c`

Comprehensive test suite with 10 unit tests covering:

1. **Functional tests:**
   - Normal matches (< 30 bytes)
   - Boundary case (29 bytes)
   - Multiple matches in single input
   - No matches scenario
   - Empty patterns
2. **Robustness tests:**
   - Oversized match truncation (82 bytes → 29 bytes)
   - Maximum payload handling (402 bytes → 29 bytes)
   - Null termination validation
   - Mixed size matches
   - Array bounds limit (10 matches max)

#### Test Results

```
[==========] tests: Running 10 test(s).
[ RUN      ] test_match_regex_normal_short_match
[       OK ] test_match_regex_normal_short_match
[ RUN      ] test_match_regex_boundary_29_bytes
[       OK ] test_match_regex_boundary_29_bytes
[ RUN      ] test_match_regex_oversized_match_truncated
[       OK ] test_match_regex_oversized_match_truncated
[ RUN      ] test_match_regex_exploit_payload_402_bytes
[       OK ] test_match_regex_exploit_payload_402_bytes
[ RUN      ] test_match_regex_multiple_matches
[       OK ] test_match_regex_multiple_matches
[ RUN      ] test_match_regex_max_matches_limit
[       OK ] test_match_regex_max_matches_limit
[ RUN      ] test_match_regex_no_matches
[       OK ] test_match_regex_no_matches
[ RUN      ] test_match_regex_null_termination
[       OK ] test_match_regex_null_termination
[ RUN      ] test_match_regex_mixed_sizes
[       OK ] test_match_regex_mixed_sizes
[ RUN      ] test_match_regex_empty_match
[       OK ] test_match_regex_empty_match
[==========] tests: 10 test(s) run.
[  PASSED  ] 10 test(s).
```

#### Manual testing

- Verified normal Rootcheck events parse correctly
- Confirmed oversized compliance tags are truncated gracefully
- Validated debug logging appears for truncation events

## Credit

Thanks to @H0j3n for reporting this bug.

## Review Checklist

- [x] Code changes reviewed
- [x] Relevant evidence provided
- [x] Tests cover the new functionality
- [x] Configuration changes documented (N/A - no changes)
- [x] Developer documentation reflects the changes (N/A - internal function)
- [x] Meets requirements and/or definition of done
- [x] No unresolved dependencies with other issues

### CODE REVIEW

File: src/analysisd/format/json_extended.c [hunk 1]
```diff
@@ -622,14 +626,29 @@ int match_regex(regex_t* r, const char* to_match, char* results[MAX_MATCHES])
...
-                sprintf(results[totalResults], "%.*s", (finish - start), to_match + start);
+                match_len = finish - start;
+                if (match_len >= MAX_STRING_LESS) {
+                    mdebug1("Match length (%d) exceeds buffer size, truncating to %d.",
+                           match_len, MAX_STRING_LESS - 1);
+                    match_len = MAX_STRING_LESS - 1;
+                }
+
+                snprintf(results[totalResults], MAX_STRING_LESS, "%.*s", match_len, to_match + start);
+                results[totalResults][MAX_STRING_LESS - 1] = '\0';
```
[the remark this answers is not in the document]
Comment (vikman90) [reply src/analysisd/format/json_extended.c:645]: This is the current behavior.

**Review by jotacarma90 — APPROVED** [review]
LGTM
